net/tcp/Kconfig

```diff
@@ -106,6 +106,27 @@ config NET_TCP_FAST_RETRANSMIT_WATERMARK
 			missing segment, without waiting for a retransmission timer to
 			expire.
 
+config NET_TCP_WINDOW_SCALE
+	bool "Enable TCP/IP Window Scale Option"
+	default n
+	---help---
+		RFC1323:
+		2. TCP WINDOW SCALE OPTION
+			The window scale extension expands the definition of the TCP
+			window to 32 bits and then uses a scale factor to carry this 32-
+			bit value in the 16-bit Window field of the TCP header (SEG.WND in
+			RFC-793).
+
+if NET_TCP_WINDOW_SCALE
+
+config NET_TCP_WINDOW_SCALE_FACTOR
+	int "TCP/IP Window Scale Factor"
+	default 7
```

Comment:
why 7?

Reply:
I just noticed that most of the network configurations are CONFIG_IOB_BUFSIZE is greater than 128, so the scale factor is set to 7. is it better to set the default value to 0?

Reply:
> I just noticed that most of the network configurations are CONFIG_IOB_BUFSIZE is greater than 128, so the scale factor is set to 7. is it better to set the default value to 0?

@anchao did you make some performance test? Maybe it is better to test it do define a good value.

Reply:
This factor only affects the sliding window size of receive and send, which does not help for performance. The window factor is best to be consistent with the iob buffer size，maybe we can calculate the scale from it.

Reply:
i guess 0 is a reasonable default because i suspect 16-bit window is just enough for the majority of setup.

maybe a more reasonable thing is to calculate the minimum value which can represent our max window. (from IOB and RECVBUF configs)

Reply:
> maybe a more reasonable thing is to calculate the minimum value which can represent our max window. (from IOB and RECVBUF configs)

as recv buffer size can change during the lifetime of the connection, maybe just from IOB configurations.

Reply:
@anchao do you want to do this "find a better scaling value" things in this PR?

Reply:
Let us do this optimization in the next PR, I think maybe you have already thought of a better solution

Reply:
ok.
then i will merge this later today unless anyone beats me or raises an objection.
(i don't like to merge a complex patch within 24 hours)